### ISSUE TITLE
Upgrade chart failed

### ISSUE THREAD

**Issue by ZhangSIming-blyq** [kubeapps/kubeapps#2930]
### Description:

I have a chart release with two node-selector, then I want to delete one and upgrade. In terminal I use helm v3 is ok, but in kubeapps dashboard it seems nothing happen, my release still have two node-selector in it.

### Steps to reproduce the issue:

1. deploy a chart
2. delete a part in value: not all failed, such as change resource limit and upgrade is ok
3. try delete a whole part, such as delete health check part; decrease a existing node-selector...

### Describe the results you received:

After delete a node-selector and upgrade, it still here in kubernetes

### Describe the results you expected:

After delete a node-selector and upgrade, It deleted truly

### Version of Helm, Kubeapps and Kubernetes:

- Output of `helm version`:

```
$ helm version
version.BuildInfo{Version:"v3.2.4", GitCommit:"0ad800ef43d3b826f31a5ad8dfbb4fe05d143688", GitTreeState:"clean", GoVersion:"go1.13.12"}

kubernetes v1.16.15
```

ANYONE have met the same problem?

**Comment by ZhangSIming-blyq** [comment]
![image](https://user-images.githubusercontent.com/58658883/120628415-3ef0b400-c497-11eb-9cd5-463f404d7404.png)

**Comment by absoludity** [comment]
Hi @ZhangSIming-blyq . I've tried to reproduce this, but wasn't able to? I installed the apache chart with:

![node-selector-1](https://user-images.githubusercontent.com/497518/120761465-8db25280-c558-11eb-9932-bb79e32df819.png)

If I then upgrade and remove the nodeSelector so it looks like:
```
nodeSelector: { }
```
and the deployment updated accordingly. I even reset it back to the original state and tried updating to:
```
nodeSelector:
```
and that also worked.

Let me know if there's more specific instructions, a specific chart, or something further we can test.
